include/bsp/iotc_bsp_tls.h

```diff
@@ -19,248 +19,219 @@
+  /** Callback context. Required by <code>iotc_bsp_tls_send_callback()</code>
+   * and <code>iotc_bsp_tls_recv_callback()</code>. */
   void* libiotc_io_callback_context;
 
-  /** Pointer to a buffer of Root CA certificate(s) in PEM format
-   * which shall be used to authenticate the IoT Core service
-   * certificate during TLS handshaking. */
+  /** Pointer to a buffer with root CA PEM certificates. */
   uint8_t* ca_cert_pem_buf;
-  /** length of the buffer containing the above certificate data */
+  /** The length, in bytes, of ca_cert_pem_buf. */
   size_t ca_cert_pem_buf_length;
 
-  /** A set of function pointers to the IoT Device SDK's memory management
-   * functions. As an option, you may provide these to your TLS implementation
-   * if you would like to track the TLS implementation's memory allocations
-   * with the optional IoTC Client's Memory Limiter functionality. */
+  /** Pointers to the client application's memory management functions. Provide
+   * these to track the TLS implementation's memory allocations with the BSP
+   * memory limiter. */
   void* (*fp_libiotc_alloc)(size_t);
   void* (*fp_libiotc_calloc)(size_t, size_t);
   void* (*fp_libiotc_realloc)(void*, size_t);
   void (*fp_libiotc_free)(void*);
 
-  /** A pointer to a NULL terminated string with the domain name that the IoT
-   * Device SDK is attempting to connect to.  Provided for use in the TLS
-   * implementation's certificate domain name check and for SNI. */
+  /** A pointer to the host's domain name. Format the domain name as a
+   * null-terminated string. */
   const char* domain_name;
 
 } iotc_bsp_tls_init_params_t;
 
 /**
  * @typedef iotc_bsp_tls_context_t
- * @brief IoT Device SDK BSP TLS's context representation type
- *
- * It is assumed that you may need to create a TLS library context
- * in the TLS BSP's initialization function.
+ * @brief TLS context representation.
  *
- * This type can be used to return such a context to the IoT Device SDK,
- * which will then provide the value to your other TLS BSP functions
- * for use in their implementations.
+ * The Device SDK doesn't read or write to this context directly. Run
+ * <code>iotc_bsp_tls_init</code> to create a TLS context.
  *
- * The IoT Device SDK is unaware of the actual content and structure
- * of this value, thus it does not read or write this context directly.
+ * @see iotc_bsp_tls_init
  */
 typedef void iotc_bsp_tls_context_t;
 
 /**
  * @function
- * @brief Provides a method for the IoT Device SDK to initialize a TLS library.
- *
- * Initializes the TLS library and returns its context through a tls_context
- * parameter.
+ * @brief Initialize a TLS library and create a TLS context.
  *
- * This function will be called before other BSP TLS functions. The
- * implementation should return IOTC_BSP_TLS_STATE_OK in case of a success or
- * IOTC_BSP_TLS_STATE_INIT_ERROR in case of a failure.
+ * Run this function first. After this function runs, it deletes the
```

Comment:
You're right. I left this sentence remained in the imperative but it should actually be in the indicative in order to tell the reader the order in which to call the functions. It doesn't match the style guide I outlined in that other comment. I updated to make it clear that this is the order in which the Device SDK calls the functions. Resolving this comment so that we can discuss in a new thread after I push the changes.